src/components/App.js

```diff
@@ -1,8 +1,196 @@
+            workScore +
+            letterPoints.find(item => {
+              return item.letter === words[i][j].toUpperCase();
+            }).point;
+      }
+    }
+    return workScore;
+    }
+  };
+  game = new Game();
+
+  const handleClick = () => {
+    // which squares contain 1's,2's,3's,.....
+    let wordN;
+    let result = '';
+    for (let i = 1; i < wordNo + 1; i++) {
+      wordN = [];
+      for (let j = 0; j < squares.length; j++) {
+        wordN = squares.filter(event => {
+          if (event.wordNums.length === 1) {
+            if (event.wordNums[0] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else if (event.wordNums.length === 2) {
+            if (event.wordNums[0] === i || event.wordNums[1] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else {
+            return false;
+          }
+        });
+      }
+      // combine words from board
+      result = '';
+      for (let k = 0; k < wordN.length; k++) {
+        result = result + wordN[k].letter;
+      }
+      words.push(result);
+    }
+    verifyBoard(words);
+  };
+
+  function checkWords() {
+    let allValid = true;
+    //let invalidWord = '';
+    for (let j = 0; j < words.length; j++) {
+      if (validWord(words[j]) === false) {
+        allValid = false;
+        invalidWord = words[j];
+        break;
+      }
+    }
+    return allValid
+  }
+
+  function verifyBoard(words) {
+    let workErrorMessage = '';
+    // Are all squares filled in?
+    for (let j = 0; j < squares.length; j++) {
+      if (squares[j].letter === '') {
+        workErrorMessage = 'Fill in all squares';
+        setErrorMessage(workErrorMessage);
+        return;
+      }
+    }
+    // are all words real words
+    if (!checkWords()) {
+      workErrorMessage = invalidWord + ' is not valid';
+    } else {
+      workErrorMessage = 'You win!!!';
+      setScore(game.calculateScore(words));
+    }
+    setErrorMessage(workErrorMessage);
+  }
+
+  function validWord(word) {
+    let wordDictionary = [];
+    if (word.length === 2) {
+      wordDictionary = wordDictionary2;
+    } else if (word.length === 3) {
+      wordDictionary = wordDictionary3;
+    } else if (word.length === 3) {
+      wordDictionary = wordDictionary4;
+    } else if (word.length === 4) {
+      wordDictionary = wordDictionary4;
+    } else {
+      wordDictionary = wordDictionary5;
+    }
+    const found = wordDictionary.find(item => {
+      return item === word.toLowerCase();
+    });
+    if (found === undefined) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
   return (
-    <div className="App">project-game18
-    </div>
+    <>
+      <h1 className="game-title">Dyna-crosswords</h1>
+      <span className="score"> Score: {score}</span>
+      <div className={(checkWords()) ? "msgValid" : "msgErr"}>{errorMessage}</div>
```

Comment:
This seems very bug prone 😨 
`checkWords` relies on the variable `words` being assigned in `handleClick` but since `words` is just a normal variable it'll be cleared next render

Reply:
checkwords replaced with a new state variable set in the code